src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,623 @@
+                .SelectMany(p => p.GetLeadingTrivia().Where(trivia => !trivia.IsDocComment() && !trivia.IsWhitespace()))
+                .AsImmutable();
+
+            // we use the class doc comment to see if we use single line doc comments or multi line doc comments
+            // if the class one isn't found, then we find the first property with a doc comment
+            // this variable doubles as a flag to see if we need to generate doc comments at all, as
+            // if it is still null, we found no meaningful doc comments anywhere
+            var exteriorTrivia = GetExteriorTrivia(typeDeclaration) ??
+                properties.SelectAsArray(GetExteriorTrivia).FirstOrDefault(t => t != null);
+
+            if (exteriorTrivia == null)
+            {
+                // we didn't find any substantive doc comments, just give the current non-doc comments
+                return SyntaxFactory.TriviaList(classTrivia.Concat(propertyNonDocComments).Select(trivia => trivia.AsElastic()));
+            }
+
+            var propertyParamComments = CreateParamComments(properties, exteriorTrivia!.Value);
+            var classDocComment = classTrivia.FirstOrNull(trivia => trivia.IsDocComment());
+            DocumentationCommentTriviaSyntax newClassDocComment;
+
+            if (classDocComment?.GetStructure() is DocumentationCommentTriviaSyntax originalClassDoc)
+            {
+                // insert parameters after summary node and the extra newline or at start if no summary
+                var summaryIndex = originalClassDoc.Content.IndexOf(node =>
+                    node is XmlElementSyntax element &&
+                    element.StartTag?.Name.LocalName.ToString() == DocumentationCommentXmlNames.SummaryElementName);
+
+                // if not found, summaryIndex + 1 = -1 + 1 = 0, so our params go to the start
+                newClassDocComment = originalClassDoc.WithContent(originalClassDoc.Content
+                    .Replace(originalClassDoc.Content[0], originalClassDoc.Content[0])
+                    .InsertRange(summaryIndex + 1, propertyParamComments));
+            }
+            else
+            {
+                // no class doc comment, if we have non-single line parameter comments we need a start and end
+                // we must have had at least one property with a doc comment
+                if (properties
+                        .SelectAsArray(p => p.GetLeadingTrivia().FirstOrNull(trivia => trivia.IsDocComment()))
+                        .Where(t => t != null)
+                        .First()?.GetStructure() is DocumentationCommentTriviaSyntax propDoc &&
+                    propDoc.IsMultilineDocComment())
+                {
+                    // add /** and */
+                    newClassDocComment = SyntaxFactory.DocumentationCommentTrivia(
+                        SyntaxKind.MultiLineDocumentationCommentTrivia,
+                        // skip initial newline, we replace it
```

Comment:
What if there isn't one?

Reply:
`CreateParamContents` always returns 2 elements for each property: a newline and a param element. We know there is at least one property because otherwise we don't activate the refactoring.

Reply:
Oh, I missed that this was processing something from a method of your creation.. I thought this was assuming that all properties would look like:

```
/// <summary>
/// hi there
/// </summary>
public int P { get; init; }
```
and not

```
/// <summary>hi there</summary>
public int P { get; init; }
```